components/fxa-client/src/telemetry.rs

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// A somewhat mixed-bag of all telemetry we want to collect. The idea is that
+// the app will "pull" telemetry via a new API whenever it thinks there might
+// be something to record.
+// It's considered a temporary solution until either we can record it directly
+// (eg, via glean) or we come up with something better.
```

Comment:
Somewhere in here, we should document out expectation that metrics will be lost if the browser closes/crashes/whatever before they are extracted by the application. I think it's a reasonable interim position for us to take, but will be good to state explicitly.

Reply:
agreed - it's worth calling out that desktop is in the same position, and we should consider if we think that's OK or not? I'd be hugely surprised/disappointed if firefox crashing explains lost metrics on desktop 😅